outline.md

```diff
@@ -16,30 +61,127 @@
+  * `concretize`: figure out a [build plan](#build-plan) for all the abstract specs to produce a list of [*concrete specs*](#abstract-vs-concrete).
+  * `install`: execute the build plan from `concretize` so all concrete specs are [*built*](#build-plan) into a *directory*.
+  * `activate`: perform [*environment modifications*](#environment-modifications) for the [concretized](#concretization) version of all abstract specs provided.
+    * This corresponds to [*global activation*][^global-activation] specifically.
+    * This abstract action covers the act of *creating a non-global filesystem view* as well.
+    * The input here may come as a *module*.
+* Spack also has many other commands which affect the output of these commands, but WOLOG they can be ignored.
+
+#### Spec syntax
+* Syntax for constraints on Spec DAGs on command line
+* Query specs, specs as "matchers"
 
-| type  | availability                                       | components                           | propagates | 
+<!-- **A spec is an index into an "environment"** -->
+
+#### Build Phases
+* Spack defines several *build phases* which may or may not execute for some [package](#package), depending on the user-selected spack command.
+  * *See [supported spack workflows](https://spack.readthedocs.io/en/latest/workflows.html).*
+
+| type  | availability                                       | components                           | propagates |
 | ---   | ---                                                | ---                                  | ---        |
 | build | build time, not runtime                            | Unknown                              | no         |
 | link  | build time, also implies run and test time for now | Unknown, probably includes a library | yes        |
 | run   | all of build, link run and test                    | Unknown                              | yes        |
 | test  | test build and run time, maybe normal build        | Unknown                              | no         |
 
-### Things not accounted for:
+* Each *build phase* acts upon a [unified environment](#unified-environment), and consists of:
+  1. applying the appropriate [*environment modifications*](#environment-modifications).
+  1. arbitrary work that modifies the *filesystem*.
 
-* PATHs required by a given phase
-* Components used, probably by phase:
-  * commands? might be covered by paths
-  * interface: headers/modules/definitions that are needed at build but not runtime
-  * libraries: possibly also static vs dynamic
-* propagation: in A->B->C does A depend on C in a given phase
-* Does it have ABI impact?
+##### Build Plan
+* A *build plan* is an ordered list of *build phases*.
+
+### Environments
+*From [Environments](https://spack.readthedocs.io/en/latest/environments.html):*
+> An environment is used to group together a set of specs for the purpose of building, rebuilding and deploying in a coherent fashion.
```

Comment:
Yeah this definition is really vague and needs fleshing out 😀

Reply:
yes, anything that has no content should be considered an implicit TODO